File: docs/snippets/all/archetypes/boxes3d_simple.cpp
// Log a single 3D box.

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_box3d");
    rec.spawn().exit_on_failure();

    rec.log("simple", dalaran::Boxes3D::from_half_sizes({{2.f, 2.f, 1.0f}}));
}

File: docs/snippets/all/archetypes/boxes3d_simple.rs
//! Log a single 3D box.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_box3d").spawn()?;

    rec.log(
        "simple",
        &dalaran::Boxes3D::from_half_sizes([(2.0, 2.0, 1.0)]),
    )?;

    Ok(())
}
